File: fabric/src/main/kotlin/cache/notification/CallResultNotification.kt
package com.obecto.perper.fabric.cache.notification
import org.apache.ignite.binary.BinaryReader
import org.apache.ignite.binary.BinaryWriter
import org.apache.ignite.binary.Binarylizable

class CallResultNotification(
    var call: String,
    var caller: String,
) : Notification(), Binarylizable {
    override fun writeBinary(writer: BinaryWriter) {
        writer.writeString("call", call)
        writer.writeString("caller", caller)
    }

    override fun readBinary(reader: BinaryReader) {
        call = reader.readString("call")
        caller = reader.readString("caller")
    }
}

File: agent/dotnet/src/Perper.Protocol/Cache/Notifications/CallResultNotification.cs
#pragma warning disable 8618, 0649
namespace Perper.Protocol.Cache.Notifications
{
    public class CallResultNotification : Notification
    {
        public string Call { get; }
        public string Caller { get; }

        public override string ToString()
        {
            return $"CallResultNotification({Call}, {Caller})";
        }
    }
}